source/MissionPanel.cpp

```diff
@@ -649,13 +687,42 @@ Point MissionPanel::DrawPanel(Point pos, const string &label, int entries) const
 		edgePos.Y() -= dy;
 	}
 
+
 	const Font &font = FontSet::Get(14);
 	pos += Point(10., 10. + (20. - font.Height()) * .5);
-	font.Draw(label, pos, selected);
+
+	// Panel sorting
+	const Color &text= *GameData::Colors().Get("medium");
+	const Sprite *arrow[2] = {SpriteSet::Get("ui/sort descending"), SpriteSet::Get("ui/sort ascending")};
+	const Sprite *checkbox[2] = {SpriteSet::Get("ui/checked"), SpriteSet::Get("ui/unchecked")};
+	const Sprite *sortIcon[4] = {SpriteSet::Get("ui/sort abc"),
+		SpriteSet::Get("ui/sort pay"), SpriteSet::Get("ui/sort speed"), SpriteSet::Get("ui/sort convenient")};
+	const Sprite *fast = SpriteSet::Get("ui/fast forward");
+
+	// Draw Sorting Columns
+	if(sorter)
+	{
+		SpriteShader::Draw(arrow[player.ShouldSortAscending()], pos + Point(SIDE_WIDTH - 25., 8.));
+
+		SpriteShader::Draw(sortIcon[player.GetAvailableSortType()], pos + Point(SIDE_WIDTH - 45., 8.));
+		if(hoverSort)
+			FillShader::Fill(pos + Point(SIDE_WIDTH - 45., 8.), Point(22., 16.), highlight);
+
+		font.Draw({"? ", {0, Alignment::RIGHT}}, pos + Point(SIDE_WIDTH - 77., 0.), text);
+		SpriteShader::Draw(checkbox[player.ShouldSortSeparatePossible()], pos + Point(SIDE_WIDTH - 70., 8.));
+
+		SpriteShader::Draw(fast, pos + Point(SIDE_WIDTH - 113., 8.));
+		SpriteShader::Draw(checkbox[player.ShouldSortSeparateDeadline()], pos + Point(SIDE_WIDTH - 100., 8.));
+	}
+
+
```

Comment:
```suggestion
```
Ik I'm being picky here but its probably because I can't find anything else that can be improved ^^